Update coordinator.py

figorr opened the pull request:
After restarting Home Assistant I found the following error at log.

> Logger: py.warnings
> Source: custom_components/ecowater_softener/sensor.py:36
> Integration: Ecowater Softener (documentation, issues)
> First occurred: 19:05:26 (1 occurrences)
> Last logged: 19:05:26
> 
> /config/custom_components/ecowater_softener/coordinator.py:52: SyntaxWarning: invalid escape sequence '\)' nextRecharge_re = "device-info-nextRecharge'\)\.html\('(?P<nextRecharge>.*)'"

I tested the change in my own installation and seems to work Ok. The error disappeared after the restart of HA.

barleybobs commented:
I am not getting this error. Could you revert your change and try restarting your home assistant instance and see if the error still occurs? The Ecowater website hasn't changed so I don't understand why this error is occurring.

figorr commented:
It is weird. You are right. I reverted the change, restarted HA ... and now the error didn't occur.

It just appeared when I updated this afternoon to the 3.3.0, coming from 3.2.0.

Maybe it was a temporary problem ... or something related to HA 2024.2.4. Because after updating the Ecowater Integration I saw this error. Then I updated coordinator.py and restarted again. After the restart I didn't see the error again, so I supposed the change was correct.  After the restart without the error i decided to update HA to 2024.2.5.  Now I reverted the change, as you suggested, and I cannot see the error under HA 2024.2.5.

Thank you @barleybobs

barleybobs commented:
No problem! Glad your issue has been fixed!

👍
